src/generators.h

```diff
@@ -124,6 +104,52 @@ struct SearchParams {
   std::variant<Whisper> inputs;
 };
 
+struct Search {
```

Comment:
rename it to Sample?

Reply:
This code does the Greedy/Beam search, the sampling is just a tiny part of it. But I might be able to hide it from generators.h